inspirehep/modules/search/api.py

```diff
@@ -0,0 +1,163 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2016 CERN.
+#
+# INSPIRE is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+from invenio_search.api import RecordsSearch
+
+
+class LiteratureSearch(RecordsSearch):
+    """Elasticsearch-dsl specialized class to search in Literature database."""
+
+    class Meta:
+        index = 'records-hep'
+        doc_types = 'hep'
```

Comment:
hmm, what do you mean? The values can be 'hardcoded' here and this will be the only place where the index and doc type is defined.

`invenio-records-rest` knows how to extract those values from the `Meta` class just by passing the search class - https://github.com/inspirehep/inspire-next/pull/1411/files#diff-3f822b17443888990c9f70f9f89d66f6R287

What use case do you have in mind?